src/undocumented_names.jl

```diff
@@ -0,0 +1,33 @@
+"""
+    test_undocumented_names(module::Module)
+
+Test that all public names in `module` have a docstring (not including the module itself).
+
+!!! tip
+    On all Julia versions, public names include the exported names.
+    On Julia versions >= 1.11, public names also include the names annotated with the `public` keyword.
+
+!!! warning
+    When running this Aqua test in Julia versions before 1.11, it does nothing.
+    Thus if you use continuous integration tests, make sure those are configured
+    to use Julia >= 1.11 in order to benefit from this test.
+"""
+function test_undocumented_names(m::Module)
+    @static if VERSION >= v"1.11"
+        # exclude the module name itself because it has the README as auto-generated docstring (https://github.com/JuliaLang/julia/pull/39093)
+        undocumented_names = filter(n -> n != nameof(m), Docs.undocumented_names(m))
```

Comment:
When having a final look at this, I stumbled over something:

IMO, this should iterate over submodules as well, but only over submodules that are part of the same package. There should be some helper functions to do that already somewhere in Aqua (e.g. https://github.com/JuliaTesting/Aqua.jl/blob/b7d973fb12e9d324441fc54da7455698acffe83b/src/exports.jl#L26-L30)

Reply:
done in the latest commit